project/OSGi.scala

```diff
@@ -65,6 +64,17 @@ object OSGi {
 
   val protobuf = exports(Seq("akka.protobuf.*"))
 
+  val protobufV3 = osgiSettings ++ Seq(
+    OsgiKeys.importPackage := Seq(
+      "!sun.misc",
+      scalaJava8CompatImport(),
+      scalaVersion(scalaImport).value,
+      configImport(),
+      "*"),
+    OsgiKeys.exportPackage := Seq("akka.protobufv3.internal.*"),
+    OsgiKeys.privatePackage := Seq("com.google.protobuf")
```

Comment:
Looks like we're embedding the `com.google.protobuf` package here. Is this intentional? Shouldn't we import these instead?

Reply:
This was indeed intentional, but it should be `google.protobuf.*` instead of `com.google.protobuf`, no?

Reply:
I think the question is rather if it is intentional to include the classes from (com.)google.protobuf in the akka protobuf jar . With your instruction you are saying, please include that package in my own jar file, but do not export it to the out side world. If there are bundles that require the google protobuf API, they won't see it from the akka bundle (again, which might be was you are intending to do). 

There is another thing that strikes me, but it might be ok: You are exporting a package that is marked as "internal". It might be required to make bundles that depend on the protobuf bundle work, but it looks strange. In our own project we never export "internal" packages as they are not for public consumption.

Perhaps you can validate that your bundle works in Karaf - then the suggested changes are ok from my side as well.

Reply:
Quick scan here, [it looks like there's JAR shading going on in the build](https://github.com/akka/akka/blob/b765835079fcb1c3af640ab7744c998ac40a2854/build.sbt#L313), in combination with the OSGi embedding, has anyone confirmed that `com.google.protobuf` is not being included twice at separate package roots?

Reply:
It was indeed intended to include the protobuf descriptors in the jar, since that was the existing behaviour. 

I did a small test of creating an akka cluster with 2 nodes each running in a different bundle and that works just fine. If needed I can provide a `features.xml` I assembled to make it run locally.